src/collapse/dtrace.rs

```diff
@@ -226,7 +231,7 @@ impl CollapsePrivate for Folder {
                 }
             }
         }
-        true
+        std::mem::take(&mut self.has_seen_digit)
```

Comment:
This doesn't seem quite right. In particular, it means `has_seen_digit` won't be reset to `false` if we hit a `return false` earlier.  Not sure if that's intentional?

Reply:
Good catch, I realized that this is more complicated than necessary.  f1e6fbffa162df3af4f45053338e465136bc40f5 simplifies it and adds a unit test.